### PR TITLE
Fixes  redirection issue for deletion of custom credential type from the details page

### PR DESCRIPTION
**Pull request by vishsanghishetty**
## Description
This PR is for [ticket](https://issues.redhat.com/browse/AAP-20602) 

**Expected:**

- [x] Delete a custom credential from its details page.
- [x] Once deleted, the page redirects to the list page of custom credential types and refreshes the list to display the modified.

**Actual:** 

- [x] Delete a custom credential from its details page.
- [x] Once deleted, the page displays React not found error.

### CODE REVIEW

File: frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx [hunk 1]
```diff
@@ -23,14 +26,12 @@ export function CredentialTypePage() {
...
     refresh,
   } = useGetItem<CredentialType>(awxAPI`/credential_types`, params.id);
-  const toolbarFilters = useCredentialTypesFilters();
-  const tableColumns = useCredentialTypesColumns();
-  const view = useAwxView<CredentialType>({
-    url: awxAPI`/credential_types/`,
-    toolbarFilters,
-    tableColumns,
+  const pageNavigate = usePageNavigate();
+  const actions = useCredentialTypeRowActions({
```
Comment (kialam) [inline frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx:30]:
```suggestion
  const actions = useCredentialTypeRowActions({
```

File: frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx [hunk 2]
```diff
@@ -23,14 +26,12 @@ export function CredentialTypePage() {
...
+    onCredentialTypesDeleted: () => pageNavigate(AwxRoute.CredentialTypes),
+    isDetailsPageAction: true,
```
Comment (kialam) [inline frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx:32]: Looks like this prop isn't being used anywhere in the hook. We can remove it and just pass a single prop here.

File: frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx [hunk 3]
```diff
@@ -23,14 +26,12 @@ export function CredentialTypePage() {
...
   });
```
Comment (kialam) [inline frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx:33]:
```suggestion
  const actions = useCredentialTypeRowActions(() => pageNavigate(AwxRoute.CredentialTypes));
```

File: frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx [hunk 4]
```diff
@@ -23,14 +26,12 @@ export function CredentialTypePage() {
...
-  const actions = useCredentialTypeRowActions(view);
+  const pageActions = actions;
```
Comment (kialam) [inline frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx:34]: Remove this

File: frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx [hunk 5]
```diff
@@ -47,7 +48,7 @@ export function CredentialTypePage() {
         ]}
         headerActions={
           <PageActions<CredentialType>
-            actions={actions}
+            actions={pageActions}
```
Comment (kialam) [inline frontend/awx/access/credential-types/CredentialTypePage/CredentialTypePage.tsx:51]:
```suggestion
            actions={actions}
```

File: frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx [hunk 6]
```diff
@@ -47,22 +45,24 @@ export function useCredentialTypeToolbarActions(view: IAwxView<CredentialType>)
...
-
-  return toolbarActions;
 }
 
-export function useCredentialTypeRowActions(view: IAwxView<CredentialType>) {
-  const { t } = useTranslation();
+export function useCredentialTypeRowActions(options: {
+  onCredentialTypesDeleted: (credentials: CredentialType[]) => void;
+  isDetailsPageAction?: boolean;
+}) {
```
Comment (kialam) [inline frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx:59]:
```suggestion
export function useCredentialTypeRowActions(onCredentialTypesDeleted: (credentialType: CredentialType[]) => void) {
```

File: frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx [hunk 7]
```diff
@@ -47,22 +45,24 @@ export function useCredentialTypeToolbarActions(view: IAwxView<CredentialType>)
...
+  const { onCredentialTypesDeleted } = options;
```
Comment (kialam) [inline frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx:60]: remove

File: frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx [hunk 8]
```diff
@@ -47,22 +45,24 @@ export function useCredentialTypeToolbarActions(view: IAwxView<CredentialType>)
...
   const pageNavigate = usePageNavigate();
-  const deleteCredentialType = useDeleteCredentialTypes(view.unselectItemsAndRefresh);
+  const deleteCredentialTypes = useDeleteCredentialTypes(onCredentialTypesDeleted);
+  const { t } = useTranslation();
 
-  const rowActions = useMemo<IPageAction<CredentialType>[]>(() => {
```
Comment (kialam) [inline frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx:65]: We'll want to keep this `rowActions` to return later

File: frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx [hunk 9]
```diff
@@ -11,21 +11,19 @@ import {
 import { useOptions } from '../../../../common/crud/useOptions';
 import { awxAPI } from '../../../common/api/awx-utils';
 import { IAwxView } from '../../../common/useAwxView';
-import { CredentialType } from '../../../interfaces/CredentialType';
 import { ActionsResponse, OptionsResponse } from '../../../interfaces/OptionsResponse';
 import { AwxRoute } from '../../../main/AwxRoutes';
 import { useDeleteCredentialTypes } from './useDeleteCredentialTypes';
+import { CredentialType } from '../../../interfaces/CredentialType';
 
 export function useCredentialTypeToolbarActions(view: IAwxView<CredentialType>) {
```
Comment (kialam) [inline frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx:19]: We can leverage passing the same on delete credential prop here as well.
```suggestion
export function useCredentialTypeToolbarActions(onCredentialTypesDeleted: (credentialType: CredentialType[]) => void) {
```

File: frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx [hunk 10]
```diff
@@ -11,21 +11,19 @@ import {
...
   const { t } = useTranslation();
   const getPageUrl = useGetPageUrl();
-
+  const deleteCredentialTypes = useDeleteCredentialTypes(view.unselectItemsAndRefresh);
```
Comment (kialam) [inline frontend/awx/access/credential-types/hooks/useCredentialTypeActions.tsx:22]:
```suggestion
  const deleteCredentialTypes = useDeleteCredentialTypes(onCredentialTypesDeleted);
```

**Review by kialam — CHANGES_REQUESTED** [review]
This is most of the way there 🥇. We can leverage props here and pass an on delete credential type handler in both toolbar and row action hooks and pass it to the `useDeleteCredentialTypes` hook. Let me know if there are any questions - I am happy to pair on this with you as well.

**Comment by vishsanghishetty** [comment]
> This is most of the way there 🥇. We can leverage props here and pass an on delete credential type handler in both toolbar and row action hooks and pass it to the `useDeleteCredentialTypes` hook. Let me know if there are any questions - I am happy to pair on this with you as well.

@kialam thanks for the feedback, much appreciated.

**Review by kialam — APPROVED** [review]
👍

**Pull request merged**